### PR TITLE
Add testing for Python custom metrics API

### PR DESCRIPTION
**Pull request by krishung5**
Python Backend: https://github.com/triton-inference-server/python_backend/pull/235

### CODE REVIEW

File: qa/python_models/custom_metrics/model.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import numpy as np
+import unittest
+import triton_python_backend_utils as pb_utils
+import requests
+
+
+class PBBLSMemoryTest(unittest.TestCase):
```
Comment (Tabrizian) [inline qa/python_models/custom_metrics/model.py:33]: I think the test needs to be renamed.

**Review by Tabrizian — COMMENTED** [review]
One minor comment, otherwise looks great!

File: qa/python_models/custom_metrics/model.py (hunk 1 above)
Comment (krishung5) [reply qa/python_models/custom_metrics/model.py]: Good catch! Fixed.

File: qa/python_models/custom_metrics/model.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+                description=
+                "test metric family with same name but different kind",
+                kind=pb_utils.GAUGE)
+            self.assertIsNone(metric_family2)
+
+        self.assertIsNotNone(metric_family1)
+
+    def test_dup_metric_family_diff_description(self):
+        # Test that a duplicate metric family name will still return the
+        # original metric family even if the description is changed
```
Comment (oandreeva-nv) [inline qa/python_models/custom_metrics/model.py:138]: Do we report this in logs? Something like: "Warning: duplicate metric family name is used, ignoring" or this is unnecessary? And do we want to log that?

**Comment by oandreeva-nv** [comment]
Nice work! Had just one question.

File: qa/python_models/custom_metrics/model.py (hunk 2 above)
Comment (rmccorm4) [reply qa/python_models/custom_metrics/model.py:138]: I don't think [MetricFamily](https://github.com/triton-inference-server/core/blob/ecbe0230d5770481d23c5f1f984ac6fac651e1df/src/metric_family.cc#L38-L62) objects are currently aware of each other, and prometheus doesn't throw any error on a (dupe name + different description) currently. 

We could probably add some static class map of name -> description if we wanted to catch this, but for now I think its OK as is.

**Review by Tabrizian — DISMISSED** [review]

File: qa/python_models/custom_metrics/config.pbtxt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+output [
+  {
+    name: "OUTPUT0"
+    data_type: TYPE_FP32
+    dims: [ 16 ]
+  }
+]
+
+instance_group [{ kind: KIND_CPU }]
```
Comment (rmccorm4) [inline qa/python_models/custom_metrics/config.pbtxt:38]: As a sanity check, can you see if everything runs OK with an instance count > 1? Since each model instance should be calling the same MetricFamily/Metric constructors/etc. - this should also test the duplicate/reference behavior.

Comment (krishung5) [reply qa/python_models/custom_metrics/config.pbtxt]: Nice thoughts! Increased the instance count.

**Review by Tabrizian — APPROVED** [review]

**Review by Tabrizian — APPROVED** [review]

**Review by rmccorm4 — APPROVED** [review]
